### PR TITLE
Eb compare functionality

### PR DESCRIPTION
**Pull request by farque65**

### CODE REVIEW

File: src/lambda_function.py [hunk 1]
```diff
@@ -163,6 +172,18 @@ def send_to_slack(val, sendto_slack_channel, sender_address):
...
                 incoming_message_request = requests.post(sendto_webhook, json=payload)
+
+                #if the slack message was not posted then send a message to sender
+                if incoming_message_request.status_code != 200:
+                    print (
+                        'In send_to_slack ephemeral Slack returned status code %s, the response text is %s' % (
+                            incoming_message_request.status_code, incoming_message_request.text)
+                    )
+
+                    send_message_to_slack('Unable to execute sendto command, retry with a valid address')
```
Comment (dnorth98) [inline src/lambda_function.py]: Should this be "retry with a valid user or channel"?

File: src/lambda_function.py [hunk 2]
```diff
@@ -189,26 +210,24 @@ def send_to_slack(val, sendto_slack_channel, sender_address):
...
                 incoming_message_request = requests.post(sendto_webhook, json=payload)
+
+                # if the slack message was not posted then send a message to sender
+                if incoming_message_request.status_code != 200:
+                    print (
+                        'In send_to_slack ephemeral Slack returned status code %s, the response text is %s' % (
+                            incoming_message_request.status_code, incoming_message_request.text)
+                    )
+
+                    send_message_to_slack('Unable to execute sendto command, retry with a valid address')
```
Comment (dnorth98) [inline src/lambda_function.py]: Same as other comment for the message

File: src/plugins/eb/__init__.py [hunk 3]
```diff
@@ -415,9 +463,131 @@ def about():
...
+	jarvis eb list applications|apps <in region/account> [sendto <slack address>]
+	jarvis eb list environments|envs <application> <in region/account> [sendto <slack address>]
+	jarvis eb describe|desc application|app <application> <in region/account> [sendto <slack address>]
+	jarvis eb describe|desc environment|env <environment> <graph> <latency|requests> <in region/account> [sendto <slack address>]
+	jarvis eb unpause|unp <environment> <in region/account> [sendto <slack address>]
+	jarvis eb compare within <region> <account> with within <region> <account> [sendto <slack address>]"""
+
+
+# retrieve json data from super jenkins for build urls
+def get_superjenkins_data(beginning_script_tag, ending_script_tag, superjenkins_link=None,superjenkins_key=None):
```
Comment (dnorth98) [inline src/plugins/eb/__init__.py]: Isn't this function copy/pasted from the ecs plugin?  Move this to a common lib..

File: src/plugins/eb/__init__.py [hunk 4]
```diff
@@ -415,9 +463,131 @@ def about():
...
+	if args.index("within") == 0:
+		if args[1] in regionList and len(args) == 3:
+			return 1
+		else:
+			return 0
+	else:
+		return 0
+
+#get the dev account role for specified account_name
+def config_get_account_rolearn(account_name, config):
```
Comment (dnorth98) [inline src/plugins/eb/__init__.py:486]: Same comment as super jenkins..is this a common function between the plugins?  If so, move it globally

File: src/plugins/eb/eb_compares/__init__.py [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+
+appversions = []
+
+def log (message):
+    print id() + ": " + message
+
+def id():
+    return "eb"
+
+def get_new_boto_session(role_arn):
```
Comment (dnorth98) [inline src/plugins/eb/eb_compares/__init__.py:13]: Common function?

File: src/plugins/eb/eb_compares/__init__.py [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+
+    if len(res) == 2 and j_tags[0] in res and j_tags[2] in res:
+        result = True
+    elif len(res) == 1 and (j_tags[0] in res or j_tags[1] in res):
+        result = True
+
+    return result
+
+
+def get_build_url(cached_array, lookup_word, prelim_version, j_tags, match_num, ismaster):
```
Comment (dnorth98) [inline src/plugins/eb/eb_compares/__init__.py:189]: Common function?

**Review by dnorth98 — CHANGES_REQUESTED** [review]

File: src/plugins/eb/eb_compares/__init__.py (hunk 6 above)
Comment (farque65) [reply src/plugins/eb/eb_compares/__init__.py:189]: This function varies in each plugin

File: src/plugins/eb/eb_compares/__init__.py (hunk 5 above)
Comment (farque65) [reply src/plugins/eb/eb_compares/__init__.py:13]: This function is only in eb

**Review by dnorth98 — APPROVED** [review]

**Pull request merged**